### DISCUSSION TITLE
请问这个文档的鼠标点击特效怎么做的，这个鼠标点击特效很惊艳啊！🙏

### DISCUSSION THREAD

**Discussion by wumingshiali** [Xiaokang2022/maliang-docs#26] [Q&A] [answered]
万分感谢

**Comment by Xiaokang2022** [comment, marked as answer]
首先，这是以 [Mkdocs](https://github.com/mkdocs/mkdocs) 为框架，[Mkdocs-Material](https://github.com/squidfunk/mkdocs-material) 为主题，深度定制的文档网站。

这里使用的鼠标点击特效采用的额外的 JavaScript 文件，也就是在配置文件 *mkdocs.yml* 中的 `extra_javascript` 里面的添加的。源代码文件名称叫 *click_colorful.js*。此文件在加载到网站后可以实现这个效果。

另外，此效果是基于一个开源项目 [click-colorful](https://github.com/ColdDay/click-colorful) 改编而来的（我记得我好像改了显示的颜色），作者是 [ColdDay](https://github.com/ColdDay)，如果您需要自己使用，请注明原作者的相关信息。

根据作者描述（见其仓库 README），此效果经过了多重优化，性能与效果兼并，即使在低性能的设备上也能有很好的效果（优化真的逆天），不像网上铺天盖地的那些劣质鼠标点击效果，因此，希望能给原作者的该项目一个 Star。

此效果不仅可以应用于 Mkdocs 文档站使用，还可以应用于博客园等网站。我最开始就是在大佬 [Kun Jinkao](https://github.com/Snoopy1866) 的博客园（链接见其 GitHub 主页）中看到了这个效果，并经过长时间搜索和发现之后，才在 GitHub 上找到了此开源项目。

**Reply by Xiaokang2022** [reply]
现在文档站以及主站的 CSS 和 JS 都迁移到另外一个仓库了，此站采用链接的方式来加载自定义 CSS 和自定义 JS，迁移到的仓库是 [Xiaokang2022.github.io](https://github.com/Xiaokang2022/Xiaokang2022.github.io)，也就是站点根目录仓库，所有相关站点的 CSS 和 JS 都集中到站点根目录方便后续的维护。

也就是说，上述点击特效的功能的 JS 可以直接通过 <https://Xiaokang2022.github.io/js/click-colorful.js> 来获取。无特殊情况，此链接不变。

相关 CSS 也可以通过链接前缀 “https://Xiaokang2022.github.io/css/” 来获取。例如需要直接获取根站点的主题和配色，那就在配置文件的 `extra_css` 加上这一个链接：<https://Xiaokang2022.github.io/css/extra.css>
